From: Joerg Roedel <jroedel@suse.de>
Date: Fri, 17 Jul 2015 10:13:48 +0200
Subject: [PATCH] kernel/modsign_uefi.c: Check for EFI_RUNTIME_SERVICES in
 load_uefi_certs
Patch-mainline: Not yet, see patches.suse/0004-MODSIGN-Import-certificates-from-UEFI-Secure-Boot.patch
References: fate#314574, bsc#856382

The load_uefi_certs() function tries to call the EFI runtime
services, but only checks for EFI_BOOT before doing so. This
causes a crash when EFI_RUNTIME_SERVICES are not available.

Fit it by explicitly checking for EFI_RUNTIME_SERVICES.

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 kernel/modsign_uefi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/modsign_uefi.c b/kernel/modsign_uefi.c
index 92d6f1c..c7b4236 100644
--- a/kernel/modsign_uefi.c
+++ b/kernel/modsign_uefi.c
@@ -145,7 +145,7 @@ static int __init load_uefi_certs(void)
 	unsigned long dbsize = 0, dbxsize = 0, moksize = 0, mokxsize = 0;
 	int ignore_db, rc = 0;
 
-	if (!efi_enabled(EFI_BOOT))
+	if (!efi_enabled(EFI_RUNTIME_SERVICES))
 		return 0;
 
 	/* See if the user has setup Ignore DB mode */
-- 
1.8.5.2

